### ISSUE TITLE
Wrong variable name when used in JSX inside a block with let or const

### ISSUE THREAD

**Issue by gaearon** [babel/babel#1584]
```js
import { Component } from 'jquery';

function hi() {
  if (true) {
    const Component = 42;
    return <Component />
  }
}
```

produces

```js
'use strict';

var _jquery = require('jquery');

function hi() {
  if (true) {
    var _Component = 42;
    return React.createElement(Component, null); // BANG! There's no Component
  }
}
```

if I remove `if` block, it works fine.

**Comment by sebmck** [comment]
Regression in 5.4.4

**Comment by sebmck** [comment]
Fixed as of 5.4.5, thanks!

**Comment by gaearon** [comment]
:+1:
